### PR TITLE
Dark mode continued

### PR DESCRIPTION
**Pull request by linusg**
Requested by #787. Based on https://github.com/owntracks/android/pull/788. Creating a separate PR as I don't have write access to the repo and therefore can't continue on the same branch/PR.

So far:

- [x] Clean up some whitespace
- [x] Fix `getDarkMode` `@export` annotation
- [x] Add German translations of new strings
- [x] Add actual values for dark theme (`Theme.Owntracks.Dark`)

### CODE REVIEW

File: project/app/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -150,4 +150,6 @@
     <string name="preferencesDeviceName">Geräte ID</string>
     <string name="preferencesHost">Hostnamen</string>
     <string name="distance_label">Abstand</string>
+    <string name="preferencesDarkMode">Dunkles Design aktivieren</string>
+    <string name="preferencesDarkModeSummary">Wechselt zu einem dunklen Design</string>
```
Comment (linusg) [inline project/app/src/main/res/values-de/strings.xml]: Here's the thing - I'd definitely say "Dak Mode aktivieren", i.e. treat that as a German word (maaaany English words have become part of spoken German in recent years).

BUT it looks like Android 10 calls its dark mode "Dunkles Design" in the German translation (see https://www.giga.de/downloads/android-p/tipps/android-dark-mode-aktivieren-so-geht-s/), so I chose that.

**Comment by linusg** [comment]
I applied the dark mode styles to most of the app. Remaining issues:

- [ ] Drawer is light - to be changed here:

  https://github.com/owntracks/android/blob/0ae08e8bb5d3da7b2968ce847ebbdbd5c6d891b7/project/app/src/main/java/org/owntracks/android/support/DrawerProvider.java#L81-L86

- [ ] Popups are light (I made an attempt to change this in `styles.xml`, didn't work :upside_down_face:)
- [ ] Long-press "tooltips" are light
- [ ] Get PreferencesActivity restart to jump back to the correct fragment

Some screenshots:

Screenshot_1581632683
Screenshot_1581632703
Screenshot_1581632709
Screenshot_1581632716
Screenshot_1581632732

**Comment by growse** [comment]
Looks really good - is that Android 29 (Q) that you've tested on?

**Comment by linusg** [comment]
> is that Android 29 (Q) that you've tested on?

I'm almost certain, but it's in the emulator - I'll have a look later.

Also regarding Android 10 - we could have an option to follow the system theme, that, when turned on, will disable the manual dark mode option.

**Comment by growse** [comment]
> > is that Android 29 (Q) that you've tested on?
> 
> I'm almost certain, but it's in the emulator - I'll have a look later.
> 
> Also regarding Android 10 - we could have an option to follow the system theme, that, when turned on, will disable the manual dark mode option.

I'd completely forgotten about the system dark mode on 10 - I guess it brings a slightly annoying layer of complexity.

Looking at the docs (https://developer.android.com/guide/topics/ui/look-and-feel/darktheme), it seems that the suggestion is that we actually provide a three-way toggle: app-specific light, app-specific dark, and whatever-the-system-says (the latter as a default).

Should this be a dropdown instead? What do other apps do here?

**Comment by linusg** [comment]
The most common pattern seems to be a radio button dialog - see https://9to5google.com/2019/10/11/android-10-apps-dark-theme/

image

**Comment by growse** [comment]
I like this. Shall I try and add to the preferences, or are you happy to look at?

**Comment by linusg** [comment]
Happy to look into it myself. My Android dev experience is a little rusty, I'll shout for help if I need.

File: project/app/src/main/res/values-de/strings.xml (hunk 1 above)
Comment (jpmens) [reply project/app/src/main/res/values-de/strings.xml:154]: I love how that article is full of "Dark Mode" and even the URL has it. ;-) 
I personally feel that "Dark Mode" makes more sense, but who am I.

Comment (linusg) [reply project/app/src/main/res/values-de/strings.xml:154]: > I personally feel that "Dark Mode" makes more sense

let's use that one then, sans quotes :smile:

**Comment by linusg** [comment]
Now that the new preferences rebuild is done I'll try to rebase and incorporate these changes in the new code.

**Comment by linusg** [comment]
Haven't been working on this for a long time and don't see that happening in the near future either, so closing for now.

**Pull request closed without merging**